<launch>
  
  
  <!-- Detection configuration -->
  <arg name="weights" default="$(find yolov8_ros)/src/n200_best.pt"/>
  <arg name="data" default="$(find yolov8_ros)/src/yolov8_tracking/yolov8/ultralytics/yolo/data/datasets/sotif.yaml"/>
  <arg name="device" default="0"/>
  
  <!-- replace imgsz -->
  <arg name="inference_size_h" default="1080"/>
  <arg name="inference_size_w" default="1920"/>


  <!-- ROS topics -->
  <arg name="input" default="/vds_node_localhost_2210/image_raw"/>
  <arg name="gt" default="/gt_5"/>
  <arg name="output_topic" default="/yolov5/detections"/>
  <arg name="bagfile" default="no1"/>
  <arg name="iter" default="1"/>


  <!-- Optional topic (publishing annotated image) -->
  <arg name="publish_image" default="false"/>
  <arg name="output_image_topic" default="/yolov5/image_out"/>


  <node name="yolov8"  pkg="yolov8_ros"  type="track.py"  output="screen" >
    <param name="weights" value="$(arg weights)"/>
    <param name="data" value="$(arg data)"/>
    <param name="device" value="$(arg device)" />
    <param name="bagfile" value="$(arg bagfile)">
    <param name="iter" value="$(arg iter)">
    
    <param name="device" value="$(arg device)" />
    <param name="input" value="$(arg input)"/>
    <param name="gt" value="$(arg gt)"/>
    <param name="output_topic" value="$(arg output_topic)"/>

    <param name="inference_size_h" value="$(arg inference_size_h)"/>
    <param name="inference_size_w" value="$(arg inference_size_w)"/>

    <param name="publish_image" value="$(arg publish_image)"/>
    <param name="output_image_topic" value="$(arg output_image_topic)"/>

  </node>
  <!-- <node pkg="rosbag" type ="play" name="rosbag_play" args="$(arg bagfile)" /> -->
   
</launch>
